### PR TITLE
Dispute blockId fix

### LINKED ISSUE #321: Error on disputing block

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Fixes dispute scenarios for rogue nodes

Fixes #321

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/dispute.go [hunk 1]
```diff
@@ -14,13 +14,15 @@ import (
...
 func HandleDispute(client *ethclient.Client, config types.Configurations, account types.Account, epoch uint32, utilsStruct UtilsStruct) error {
-	numberOfProposedBlocks, err := utilsStruct.razorUtils.GetNumberOfProposedBlocks(client, account.Address, epoch)
+	sortedProposedBlockIds, err := utilsStruct.razorUtils.GetSortedProposedBlockIds(client, account.Address, epoch)
 	if err != nil {
-		log.Error(err)
 		return err
 	}
-	for i := 0; i < int(numberOfProposedBlocks); i++ {
-		proposedBlock, err := utilsStruct.razorUtils.GetProposedBlock(client, account.Address, epoch, uint8(i))
+	log.Info("SortedProposedBlockIds: ", sortedProposedBlockIds)
```
Comment (ashish10677) [inline cmd/dispute.go:21]: Make this as log.Debug

File: cmd/dispute_test.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func TestHandleDispute(t *testing.T) {
 		{
 			name: "Test 1: When HandleDispute function executes successfully when there is a dispute case",
 			args: args{
-				numberOfProposedBlocks: 4,
+				sortedProposedBlockIds: []uint8{3, 1, 2},
```
Comment (ashish10677) [inline cmd/dispute_test.go:212]: Can you add more number of blocks and more realistic median values?

**Review by ashish10677 — CHANGES_REQUESTED** [review]

**Review by SkandaBhat — APPROVED** [review]

**Comment by SkandaBhat** [comment]
Add couple of more tests here. Coverage has decreased.

**Comment by ashish10677** [comment]
@Yashk767 build is failing

**Comment by Yashk767** [comment]
> Add couple of more tests here. Coverage has decreased.

Coverage for dispute is covered same as before. As we have added a sortedProposedBlock function in utils so its decreasing for utils.

File: cmd/dispute_test.go [hunk 3]
```diff
@@ -209,12 +209,12 @@ func TestHandleDispute(t *testing.T) {
 		{
 			name: "Test 1: When HandleDispute function executes successfully when there is a dispute case",
 			args: args{
-				numberOfProposedBlocks: 4,
+				sortedProposedBlockIds: []uint8{3, 1, 2, 5, 4},
 				proposedBlock: bindings.StructsBlock{
-					Medians: []uint32{100, 200, 300},
+					Medians: []uint32{6901548, 498307},
```
Comment (ashish10677) [inline cmd/dispute_test.go:214]: ActiveAssetIds is 3, but median is only 2?

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/dispute_test.go (hunk 3 above)
Comment (Yashk767) [reply cmd/dispute_test.go:214]: Ohh yes, I will change it.

**Review by ashish10677 — APPROVED** [review]
LGTM

**Pull request merged**